todos/src/components/EditTask.js

```diff
@@ -3,20 +3,39 @@ import del from "icons/delete.svg";
 import done from "icons/done.svg";
 import "styles/addTask.css";
 import { useDispatch } from "react-redux";
-import { editUpdateCompleted, updateTodo } from "actions/index";
-import { successMessage, infoMessage } from "toastMethods";
 import {
-  STATE_CHANGE_MESSAGE,
-  EDIT_CANCEL_MESSAGE,
-  EDIT_SUCCESS_MESSAGE,
-} from "constants";
+  editTodo,
+  editUpdateCompleted,
+  updateTodo,
+  selectedCardId,
+} from "actions/index";
+import { infoMessage } from "toastMethods";
+import { EDIT_CANCEL_MESSAGE } from "constants";
+import { useSelector } from "react-redux";
+import spinner from "icons/spinner.svg";
 
 const EditTask = (props) => {
   const [inputData, setInputData] = useState("");
   const dispatch = useDispatch();
+
+  const doneHandlerHelper = (data) => {
+    dispatch(selectedCardId(props.id));
+    if (data !== props.currentData) {
+      dispatch(
+        updateTodo(props.id, data, props.onEdit, true),
+        setInputData("")
+      );
+    } else {
+      dispatch(
+        updateTodo(props.id, props.currentData, props.onEdit, false),
+        setInputData("")
+      );
+    }
```

Comment:
Can we use early return here?

Reply:
@hhu-wtag Bhai, Although using early return makes this code efficient it might break my functionalities here.
we need to call updateTodo in both `data !== props.currentData` and `data === props.currentData` conditions. cause If the two values are not equal, we should pass `true` as the last argument, and if they are equal, we should pass `false`.

However, I believe we can make the code more efficient by avoiding the if-else statement and writing it like this:

```
  const doneHandlerHelper = (data) => {
    dispatch(selectedCardId(props.id));
    const isDataSameAsCurrentData = data === props.currentData;
    dispatch(
      updateTodo(
        props.id,
        isDataSameAsCurrentData ? props.currentData : data,
        props.onEdit,
        !isDataSameAsCurrentData
      ),
      setInputData("")
    );
  };
```

What do you think, @hhu-wtag Bhai?

Reply:
We shouldn't compare objects using `===` because it's an equality check of function reference rather than the actual data. If early return causes complications we can resort to the current solution as well.

Reply:
Okay, Bhai. Restored.